indispensable-josef/src/commonMain/kotlin/at/asitplus/signum/indispensable/josef/JwsHeader.kt

```diff
@@ -52,6 +52,40 @@ data class JwsHeader(
     @SerialName("typ")
     val type: String? = null,
 
+    /**
+     * https://www.rfc-editor.org/rfc/rfc7515.html
+     *
+     * 4.1.11.  "crit" (Critical) Header Parameter
+     *
+     *    The "crit" (critical) Header Parameter indicates that extensions to
+     *    this specification and/or [JWA] are being used that MUST be
+     *    understood and processed.  Its value is an array listing the Header
+     *    Parameter names present in the JOSE Header that use those extensions.
+     *    If any of the listed extension Header Parameters are not understood
+     *    and supported by the recipient, then the JWS is invalid.  Producers
+     *    MUST NOT include Header Parameter names defined by this specification
+     *    or [JWA] for use with JWS, duplicate names, or names that do not
+     *    occur as Header Parameter names within the JOSE Header in the "crit"
+     *    list.  Producers MUST NOT use the empty list "[]" as the "crit"
+     *    value.  Recipients MAY consider the JWS to be invalid if the critical
+     *    list contains any Header Parameter names defined by this
+     *    specification or [JWA] for use with JWS or if any other constraints
+     *    on its use are violated.  When used, this Header Parameter MUST be
+     *    integrity protected; therefore, it MUST occur only within the JWS
+     *    Protected Header.  Use of this Header Parameter is OPTIONAL.  This
+     *    Header Parameter MUST be understood and processed by implementations.
```

Comment:
This would also concern header parameter `alg`, since `alg` may also implicitly require other header parameters, right?
At least that is my understanding according to the specification regarding jws validation:
```
  5.  Verify that the implementation understands and can process all
       fields that it is required to support, whether required by this
       specification, **by the algorithm being used**, or by the "crit"
       Header Parameter value, and that the values of those parameters
       are also understood and supported.
```

Reply:
@JesusMcCloud opinions on this? i would like to validate this as part of JWT (signature) validation (in `supreme`)